### PR TITLE
Remove dead source files in vm\**

### PR DESCRIPTION
**Pull request by danmosemsft**
Contributes to https://github.com/dotnet/coreclr/issues/9550

All the cpp files removed were not being compiled (not present in cmakelists.txt). I have only removed headers if (1) they were not included anywhere or (2) contained nothing but declarations for the deleted impl or (3) I inlined what was left.

Several of these will go away anyway when I finish removing dead defines.

I didn't look at dead files outside the vm tree.

@jkotas

### CODE REVIEW

File: src/vm/mda.cpp [hunk 1]
```diff
@@ -1,4017 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
```
Comment (stephentoub) [inline src/vm/mda.cpp]: Do we have any plans to bring back MDAs?  I've found at least some of them to be useful in the past, e.g. stack imbalance helping to highlight incorrectly written P/Invokes.

**Comment by jkotas** [comment]
Delete matching header files as well, e.g. objectclone.h, remoting.h, ... ?

**Comment by jkotas** [comment]
I have read your comment now that you were not deleting some of them. I think it should be pretty easy to just delete them now, but if you would like to do it later, it is fine too.

File: src/vm/mda.cpp (hunk 1 above)
Comment (jkotas) [inline src/vm/mda.cpp]: I agree that we may want to consider bringing some of the MDAs back - it is why I have not checked `FEATURE_MDA` in https://github.com/dotnet/coreclr/issues/9067.

**Comment by danmosemsft** [comment]
@mmitche hardening needed?
https://ci.dot.net/job/dotnet_coreclr/job/master/job/checked_windows_nt_bld_prtest/4250/consoleText
```
ERROR: Failed to notify endpoint with id 'HTTP:legacy-notification-url'
java.net.SocketTimeoutException: Read timed out
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465)
	at sun.security.ssl.InputRecord.read(InputRecord.java:503)
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:973)
	at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:930)
	at sun.security.ssl.AppInputStream.read(AppInputStream.java:105)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
	at sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:704)
	at sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:647)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1569)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1474)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:338)
	at com.tikal.hudson.plugins.notification.Protocol$3.send(Protocol.java:115)
	at com.tikal.hudson.plugins.notification.Phase.handle(Phase.java:47)
	at com.tikal.hudson.plugins.notification.JobListener.onStarted(JobListener.java:31)
	at hudson.model.listeners.RunListener.fireStarted(RunListener.java:240)
	at hudson.model.Run.execute(Run.java:1716)
	at hudson.model.FreeStyleBuild.run(FreeStyleBuild.java:43)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:404)
```

**Comment by mmitche** [comment]
@danmosemsft Nope. Doesn't affect the result of the job.

**Comment by danmosemsft** [comment]
@mmitche I agree that wsn't why the job failed. Just noticed it.

**Comment by danmosemsft** [comment]
@dotnet-bot test windows_nt x64 debug build and test (https://github.com/dotnet/coreclr/issues/9388)

**Comment by danmosemsft** [comment]
@jkotas I just noticed you didn't actually sign off. My bad. LMK if you need changes.

**Pull request merged**